### PR TITLE
Move make migrations.lock action into makemigrations management command

### PR DESCRIPTION
**Pull request by joxl**
## Technical Summary

Updating the `migrations.lock` file is now integrated with the Django `makemigrations` management command rather than requiring developers to run a separate `make migrations.lock` command every time a new migration is created.

This new functionality writes an updated lock file any time a developer uses `makemigrations` to generate a new migration. Two new "action options" have also been added to the `makemigrations` command for convenience:
- `--lock-check` Exit with a non-zero status if the lock file is inconsistent.
- `--lock-update` Write a new (updated) lock file and exit.

Note that the `--lock-update` action is only necessary in scenarios where a developer manually creates a migration file without using the manage command, or if a developer decides to rename an existing migration module after it is created.

The `manage.lock` file no longer includes the `[X]` prefix because it is not needed and leaving it would cause problems since the lock file is now updated when a migration is created (which will always be before it is applied). The lock file also now contains a comment header explaining how it gets updated and how to refresh it manually.  The combination of the automatic updates (via `makemigrations`) and the comment header ensures that a developer should be unable to create a new migration without being able to quickly identify how to generate a consistent `migrations.lock` file.

Extends previous functionality added in #30789.

Jira [SAAS-12801](https://dimagi-dev.atlassian.net/browse/SAAS-12801)

### Safety story

Changes only affect tests and code contributors.

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

There are no real migrations in this PR, only a new `migrations.lock` file, which will automatically trigger the migrations label.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

File: hqscripts/management/commands/makemigrations.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            file.write("".join(self.get_migrations_list()))
+
+    def check_migrations_lock(self):
+        """Check if the current migrations list matches the migrations.lock."""
+        with open(self.lock_path, "r") as file:
+            frozen = list(file)
+        current = self.get_migrations_list()
+        if frozen == current:
+            return True
+        return False
```
Comment (millerdev) [inline hqscripts/management/commands/makemigrations.py:97]: nit
```suggestion
        return frozen == current
```

File: hqscripts/management/commands/makemigrations.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                         returned migration list.
+        """
+        stream = StringIO()
+        command = showmigrations.Command(stdout=stream, no_color=True)
+        # Use `run_from_argv()` to avoid extra command setup boilerplate.
+        # First two argv items are `prog` and `subcommand`, used to setup the
+        # command's argument parser, leaving them empty because it works.
+        command.run_from_argv(["", "", "--list"])
+        stream.seek(0)
+        lines = [f"{line}\n" for line in preamble.split("\n")]
```
Comment (millerdev) [inline hqscripts/management/commands/makemigrations.py:77]: nit: Python has a function for that
```suggestion
        lines = preamble.splitlines(keepends=True)
```

**Review by millerdev — APPROVED** [review]
Excellent! I love that existing `makemigrations` workflows have not changed.

File: hqscripts/management/commands/makemigrations.py (hunk 1 above)
Comment (joxl) [reply hqscripts/management/commands/makemigrations.py:97]: There you go finding the brain freezes again... :man_facepalming:

File: hqscripts/management/commands/makemigrations.py (hunk 2 above)
Comment (joxl) [reply hqscripts/management/commands/makemigrations.py:77]: Ohhhhh, TIL!

**Review by millerdev — APPROVED** [review]

**Pull request merged**